contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -249,6 +252,20 @@ boolean LookupAccountSid(String lpSystemName, PSID Sid,
     boolean IsValidSid(PSID pSid);
 
     /**
+     * he EqualSid function tests two security identifier (SID) values for equality.
+     * Two SIDs must match exactly to be considered equal.
+     * @param pSid1
+     *             A pointer to the first SID structure to compare. This structure is assumed to be valid.
+     * @param pSid2
+     *             A pointer to the second SID structure to compare. This structure is assumed to be valid.
+     * @return If the SID structures are equal, the return value is nonzero.
+     *         If the SID structures are not equal, the return value is zero. To get extended error
+     *         information, call GetLastError.
+     *         If either SID structure is not valid, the return value is undefined.
+     */
+    boolean EqualSid(PSID pSid1, PSID pSid2);
```

Comment:
Looks to me that it compares class types then calls Pointer#equals which compares that the native peer memory address is the same.  EqualSid compares the _contents_ under those memory address.